commands/clone.go

```diff
@@ -116,9 +135,32 @@ func transformCloneArgs(args *Args) {
 
 				url := project.GitURL(name, owner, isSSH)
 				args.ReplaceParam(i, url)
-			}
 
-			break
+				targetDir = name
+				if repo.Parent != nil && args.Command != "submodule" {
+					args.AfterFn(func() error {
+						upstreamName := "upstream"
+						if originName == "upstream" {
+							upstreamName = "origin"
+						}
+						upstreamUrl := project.GitURL(repo.Parent.Name, repo.Parent.Owner.Login, repo.Parent.Private)
+						addRemote := cmd.New("git")
+						addRemote.WithArgs("-C", targetDir)
+						addRemote.WithArgs("remote", "add", "-f", upstreamName, upstreamUrl)
+						if !quiet {
+							ui.Errorf("Adding remote '%s' for the '%s/%s' repo\n",
+								upstreamName, repo.Parent.Owner.Login, repo.Parent.Name)
+						}
+						output, err := addRemote.CombinedOutput()
+						if err != nil {
+							ui.Errorln(output)
+						}
+						return err
```

Comment:
Maybe it should do a `git fetch` for the added remote at this point? The normal case for doing a `git clone` fetches the full history for the cloned repo; since we're basically cloning two repos at once here, maybe we should do fetch so we have both repos' history.

Reply:
Notice that we're doing `git remote add -f`. The extra flag automatically initiates a fetch.

Reply:
Gotcha. Missed that.